### ISSUE TITLE
Using CoGeoTiff with the compiled (non module) version of OL

### ISSUE THREAD

**Issue by pitommasi** [openlayers/openlayers#14952]
I've downloaded the compiled script from the download package but can't get the [CoGeotiff example](https://openlayers.org/en/latest/examples/cog.html) to work:
I have succesfully converted the import and the module syntax to make the basic map working, e.g.:

```
const Map = ol.Map;
const View = ol.View;
const TileLayer = ol.layer.Tile;
const XYZ = ol.source.XYZ;

new Map({
	target: 'map',
	layers: [
		new TileLayer({
			source: new XYZ({
				url: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png'
			})
		})
	],
	view: new View({
		center: [0, 0],
		zoom: 2
	})
});
```

however, I'm having issues with the GeoTiff part, I tried converting

`import GeoTIFF from 'ol/source/GeoTIFF.js';`

to 

`const GeoTIFF = ol.source.GeoTIFF;`

but then I get 

GeoTIFF.js:244 Uncaught TypeError: GeoTIFF.fromUrl is not a function

So I guess that's not the way to go... 
I'm guessing I might have the same issues with 

`import TileLayer from 'ol/layer/WebGLTile.js';`

So how can I work with GeoTIFF using the compiled version of OL?

**Comment by mike-000** [comment]
You will need `geotiff.js` as well as `ol.js`
```
    <script src="https://cdn.jsdelivr.net/npm/geotiff@2.0.7/dist-browser/geotiff.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/ol@v7.4.0/dist/ol.js"></script>

```
`ol.source.GeoTIFF` needs to be used in an `ol.layer.WebGLTile` layer

```
const TileLayer = ol.layer.WebGLTile;
const GeoTIFF = ol.source.GeoTIFF;
```

**Comment by pitommasi** [comment]
thanks, but that's what I've done: this is in my html file:

`    <script src="https://cdn.jsdelivr.net/npm/geotiff@2.0.7/dist-browser/geotiff.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/ol@v7.4.0/dist/ol.js"></script>`
	
and this is the js code:

`const GeoTIFF = ol.source.GeoTIFF;

const source = new GeoTIFF({
	sources: [
		{
			url: 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/Q/WD/2020/7/S2A_36QWD_20200701_0_L2A/TCI.tif',
		}
	],
});`

I still get:

Uncaught TypeError: GeoTIFF.fromUrl is not a function

**Comment by pitommasi** [comment]
this is the ol.js row that's causing it:

`    request = tiffFromUrl(source.url, options);`

**Comment by mike-000** [comment]
I get the same error with that example, but other examples are working, so it may be a bug.

**Comment by ahocevar** [comment]
@pitommasi By assigning `const GeoTIFF = ol.source.GeoTIFF`, you're re-assigning the global `GeoTIFF` variable that is the container for geotiff.js. Create an `ol.source.GeoTIFF` instance directly instead:
```js
const source = new ol.source.GeoTIFF({
  sources: [
    {
      url: 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/Q/WD/2020/7/S2A_36QWD_20200701_0_L2A/TCI.tif',
    }
  ],
})
```

**Comment by ahocevar** [comment]
Working example: https://jsfiddle.net/mwLq3efy/3/

**Comment by pitommasi** [comment]
![image](https://github.com/openlayers/openlayers/assets/17688589/6aed1052-5e3a-48d3-9290-d9ec7e7eed52)

**Comment by mike-000** [comment]
@pitommasi had you deleted the `const GeoTIFF = ol.source.GeoTIFF;` when you got that error?

Another way to avoid the name conflict would be to wrap the code in an extra set of `{ }`

```
{
  const GeoTIFF = ol.source.GeoTIFF;
  const Map = ol.Map;
  const TileLayer = ol.layer.WebGLTile;

  const source = new GeoTIFF({
    sources: [
      {
        url: 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/Q/WD/2020/7/S2A_36QWD_20200701_0_L2A/TCI.tif',
      },
    ],
  });

  const map = new Map({
    target: 'map',
    layers: [
      new TileLayer({
        source: source,
      }),
    ],
    view: source.getView(),
  });
}
```

**Comment by pitommasi** [comment]
> @pitommasi had you deleted the `const GeoTIFF = ol.source.GeoTIFF;` when you got that error?

Yes, I have.

I had a similar line for Map, though:
`const Map = ol.Map;`

but funnily enough, that was working until I've started adding the Geotiff file...
Now everything works, thanks to both for the help.

So, to recap:

I had to remove both

`const Map = ol.Map;` although this wasn't causing issues per se but only after adding the geotiff source initialisation code

and  

`const GeoTIFF = ol.source.GeoTIFF`

and 

I've used

`const source = new ol.source.GeoTIFF({
  sources: [
    {
      url: 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/Q/WD/2020/7/S2A_36QWD_20200701_0_L2A/TCI.tif',
    }
  ],
})`

to initialise the source

**Comment by pitommasi** [comment]
So now I have a doubt, could these lines create issues?

`const View = ol.View;
const TileLayer = ol.layer.Tile;
const XYZ = ol.source.XYZ;`

**Comment by mike-000** [comment]
`const Map = ol.Map` would override the javascript Map constructor https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map unless you put your code in its own namespace using extra brackets or a function.  Similarly `const Image = ol.layer.Image` would break the javascript Image constructor https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/Image

**Comment by pitommasi** [comment]
Right! Somehow I was so concentrated on OL that wasn't considering possible conflicts with native JS  :-P

**Comment by pitommasi** [comment]
Thanks again for helping.
Not wanting to set up a build porcess for now, is there ny caveat I should consider in using the compiled versions of OL and GeoTIFF?

**Comment by ahocevar** [comment]
@pitommasi The full build contains everything one could ever need of OpenLayers and GeoTIFF. So your browser has to load, parse and just-in-time compile way more code than with a build, which only contains code that the application needs.

**Comment by pitommasi** [comment]
I thought so, so besides that, no issues, right?
